src/test/java/dev/blaauwendraad/masker/json/LargeStreamTest.java

```diff
@@ -0,0 +1,86 @@
+    static final String ACTUAL_RESULT_FILE_NAME = "actual_result.json";
+
+    @BeforeAll
+    static void setupFiles() throws IOException {
+        // prepare input.json file and expected_result.json
+        try (FileWriter inputFileWriter = new FileWriter(INPUT_FILE_NAME);
+             FileWriter expectedResultFileWriter = new FileWriter(EXPECTED_RESULT_FILE_NAME)) {
+            String line = "{\"key\":\"mask\"},".repeat(1024);
+            String expectedResultLine = "{\"key\":\"***\"},".repeat(1024);
+            inputFileWriter.write("[");
+            expectedResultFileWriter.write("[");
+            for (int i = 0; i < 65536; i++) {
+                inputFileWriter.write(line);
+                expectedResultFileWriter.write(expectedResultLine);
+                inputFileWriter.flush();
+                expectedResultFileWriter.flush();
+            }
+            inputFileWriter.write("{\"key\":\"mask\"}");
+            expectedResultFileWriter.write("{\"key\":\"***\"}");
+            inputFileWriter.write("]");
+            expectedResultFileWriter.write("]");
+            inputFileWriter.flush();
+            expectedResultFileWriter.flush();
+        }
+    }
+
+    @Test
+    @Disabled("Run it manually")
+    void shouldProcessLargeJson() throws IOException {
+        // process the input file and write the result into actual_result.json file
+        try (FileInputStream inputStream = new FileInputStream(INPUT_FILE_NAME);
+             FileOutputStream outputStream = new FileOutputStream(ACTUAL_RESULT_FILE_NAME)) {
+            JsonMasker jsonMasker = JsonMasker.getMasker(JsonMaskingConfig.builder().allowKeys().build());
+            jsonMasker.mask(inputStream, outputStream);
+        }
+
+        // assert correctness of the result
+        try (FileInputStream expectedResultFile = new FileInputStream(EXPECTED_RESULT_FILE_NAME);
+             FileInputStream actualResultFile = new FileInputStream(ACTUAL_RESULT_FILE_NAME)) {
+            String expected = new String(expectedResultFile.readNBytes(512), StandardCharsets.UTF_8);
+            String actual = new String(actualResultFile.readNBytes(512), StandardCharsets.UTF_8);
+            while (!expected.isEmpty() || !actual.isEmpty()) {
+                Assertions.assertEquals(expected, actual);
+                expected = new String(expectedResultFile.readNBytes(512), StandardCharsets.UTF_8);
+                actual = new String(actualResultFile.readNBytes(512), StandardCharsets.UTF_8);
+            }
+            Assertions.assertEquals(0, expectedResultFile.available());
+            Assertions.assertEquals(0, actualResultFile.available());
+        }
+    }
+
+    @AfterAll
+    static void cleanUpFiles() throws IOException {
+        // clean up files
+        Files.deleteIfExists(Path.of(INPUT_FILE_NAME));
+        Files.deleteIfExists(Path.of(EXPECTED_RESULT_FILE_NAME));
+        Files.deleteIfExists(Path.of(ACTUAL_RESULT_FILE_NAME));
+    }
```

Comment:
There's nice `@TempFile` extension from JUnit that does that automatically:
```
@TempFile Path inputFile;
@TempFile Path expectedResultFile;
@TempFile Path actualResultFile;
```

Reply:
Nice one, didn't know about this

Reply:
Done